pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -605,6 +605,15 @@ func (s *kafkaSink) handleBufferedRetries(msgs []*sarama.ProducerMessage, retryE
 		// SendMessages will attempt to send all messages into an AsyncProducer with
 		// the client's config and then block until the results come in.
 		lastSendErr = newProducer.SendMessages(msgs)
+		if lastSendErr != nil {
+			// nolint:errcmp
+			if sendErrs, ok := lastSendErr.(sarama.ProducerErrors); ok && len(sendErrs) > 0 {
+				// Just check the first error since all these messages being retried
+				// were likely from a single partition and therefore would've been
+				// marked with the same error.
+				lastSendErr = sendErrs[0].Err
```

Comment:
I assume you've verified this in a live cluster; do you mind pasting log messages as this logic works its way through?